docroot/modules/custom/foia_migrate/src/data-groomer.php

```diff
@@ -300,3 +301,113 @@ function remove_departments_property_from_agencies(array &$agency_data) {
     unset($agency->departments);
   }
 }
+
+/**
+ * Extract that last item in the Request Times Stats data object.
+ *
+ * @param array &$components
+ *   Array of all components.
+ */
+function extract_processing_data(array &$components, array $agency_data) {
+  foreach ($components as &$component) {
+    if (isset($component->request_time_stats)) {
+      $latest_data = end($component->request_time_stats);
+      $year = key($component->request_time_stats);
+      $component->latest_request_time_stats = $latest_data;
+      $component->latest_request_time_stats_year = $year;
+    }
+    else {
+      $latest_data = extract_processing_data_from_agency($component->agency_name, $agency_data);
+      if ($latest_data) {
+        $component->latest_request_time_stats = $latest_data['data'];
+        $component->latest_request_time_stats_year = $latest_data['year'];
+      }
+    }
+
+    // Add empty values for any statistics missing data.
+    $latest_request_time_stats_array = (array) $component->latest_request_time_stats;
+    $component->latest_request_time_stats = (object) insert_empty_days_values($latest_request_time_stats_array);
```

Comment:
Does the code not work as expected without the casts to `array` and `object`?

Reply:
When I tried to send `$component->latest_request_time_stats` as an argument to the `insert_empty_days_values` function I got: `Catchable fatal error: Argument 1 passed to insert_empty_days_values() must be an instance of object, instance of stdClass given`

Reply:
Oh interesting. Hmm, if that's the case I would imagine this works?
```
$latest_request_time_stats = (array) $component->latest_request_time_stats;
$component->latest_request_time_stats = insert_empty_days_values($latest_request_time_stats);
```
(Removed the `(object)`)